python/paddle/fluid/dygraph/dygraph_to_static/partial_program.py

```diff
@@ -155,6 +164,36 @@ def _set_grad_type(self, params):
                 continue
             param._set_grad_type(grad_var.type())
 
+    def _check_params_all_inited(self, main_program):
+        """
+        Check all params from main program are already initialized, see details as follows:
+            1. all parameters in self._params should be type `framework.ParamBase` which are created in dygraph.
+            2. all parameters from transformed program can be found in self._params.
+               Because they share same data with ParamBase of original dygraph.
+        """
+        if not isinstance(self._params, (list, tuple)):
+            raise TypeError(
+                "Type of self._params should be list or tuple, but received %s."
+                % type(self._params))
+
+        params_name_set = set()
+        for i, param in enumerate(self._params):
+            if not isinstance(param, framework.ParamBase):
+                raise TypeError(
+                    'Type of self._params[{}] shoule be framework.ParamBase, but received {}.'.
+                    format(i, type(param)))
+            params_name_set.add(param.name)
+
+        for block in main_program.blocks:
+            for name, var in block.vars.items():
+                if isinstance(var, framework.Parameter):
+                    if name not in params_name_set:
+                        raise ValueError(
+                            "We don't support to define layer with parameters in the function "
```

Comment:
Here you explain that "we don't do so" but not why. Could you add the reason to users why they have to do it? Just like our chat in this morning:

"We don't support to define layer with parameters in the function decorated by `@declarative` because that will re-define parameters every time when you run the function. We found parameter(%s) created in the decorated function. Please define the parameter layer in in `__init__` function."

This is good for both SWEs and users because they know background why we check the parameter is in __init__ here.

Reply:
Thanks, the error message will be output as followed:
```
ValueError: 
	We don't support to define layer with parameters in the function decorated by `@declarative`.
	Because that will re-defined parameters every time when you run the function.
	But we found parameter(linear_0.w_0) was created in the decorated function.
	Please define the layer with parameters in `__init__` function.
```